### ISSUE TITLE
Closing interactive shell prevents terminal to wait for input

### ISSUE THREAD

**Issue by eduardtudor** [elixir-lang/elixir#6966]
### Environment

* Elixir & Erlang versions (elixir --version): Elixir 1.5.2

### Current behavior

When closing the interactive shell the next program will be prevented to wait for input

```
[user@host~]$ iex
Erlang/OTP 20 [erts-9.0.1] [source] [64-bit] [smp:2:2] [ds:2:2:10] [async-threads:10] [hipe] [kernel-poll:false]

Interactive Elixir (1.5.2) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> 
BREAK: (a)bort (c)ontinue (p)roc info (i)nfo (l)oaded
       (v)ersion (k)ill (D)b-tables (d)istribution
^C[user@host ~]$ sudo pacman -S vim
resolving dependencies...
looking for conflicting packages...

Packages (2) vim-runtime-8.0.1176-1  vim-8.0.1176-1

Total Download Size:    6.59 MiB
Total Installed Size:  30.16 MiB

:: Proceed with installation? [Y/n] [user@host~]$
```

**Comment by josevalim** [comment]
@eduardtudor can you consistently reproduce the issue or was it a temporary behaviour?

**Comment by josevalim** [comment]
Btw, iex shells does consume all input but I wouldn't expect it to still do so after the iex process is done. It may be that the process has not fully terminated for some reason or there is another process hanging in the background. You can verify that by running `ps aux | grep beam`.

**Comment by eduardtudor** [comment]
@josevalim I can consistently reproduce the issue for the same session, no result for `ps aux | grep beam` related to beam

**Comment by eduardtudor** [comment]
I just googled other way of exiting the iex, the issue only exist for Ctrl+C + Ctrl+C and Ctrl+C + abort. I tried Ctrl + | and the issue is not happening.

**Comment by josevalim** [comment]
@eduardtudor ok, so that is definitely an Erlang behaviour. Can you try to reproduce it in another way so we can make a proper report? Does it happen if you start `cat` immediately after? Also, what is your Erlang version?

**Comment by eduardtudor** [comment]
I get  `cat: -: Resource temporarily unavailable` 

Erlang/OTP 20

**Comment by josevalim** [comment]
I can't reproduce it on Mac. :( Can you please try the same thing by using `erl` instead of `iex`? Then you can also use Ctrl+C twice to exit.

**Comment by eduardtudor** [comment]
I can also reproduce using `erl`

**Comment by eduardtudor** [comment]
So I researched a little and found that maybe the interactive shell leaves the stdin in non blocking

Program to set the stdin back to blocking
```
#include <stdio.h>
#include <unistd.h>
#include <fcntl.h>

int main() {

  int flags = fcntl (0, F_GETFL);
  if (fcntl(0, F_SETFL, flags & ~O_NONBLOCK))
     printf("Failed to make stdin blocking");
}
```
I ran the program after exiting the shell using Ctrl+C twice and pacman was able to wait for input.
```
[user@host ~]$ iex
Erlang/OTP 20 [erts-9.0.1] [source] [64-bit] [smp:2:2] [ds:2:2:10] [async-threads:10] [hipe] [kernel-poll:false]

Interactive Elixir (1.5.2) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> 
BREAK: (a)bort (c)ontinue (p)roc info (i)nfo (l)oaded
       (v)ersion (k)ill (D)b-tables (d)istribution
^C[user@host ~]./a.out
[user@host f ~]$ sudo pacman -S xterm
warning: xterm-330-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) xterm-330-1

Total Installed Size:  0.81 MiB
Net Upgrade Size:      0.00 MiB

:: Proceed with installation? [Y/n] n
[user@host  ~]$ 
```

**Comment by josevalim** [comment]
Thanks @eduardtudor. That's definitely an Erlang bug then. You can check if it is fixed on Erlang 20.1 but, if not, I would recommend you to file a report here: bugs.erlang.org

**Comment by eduardtudor** [comment]
Ok will do, thanks!
